cadence/transactions/evm/create_new_account_with_coa.cdc

```diff
@@ -0,0 +1,64 @@
+import Crypto
+
+import "EVM"
+
+/// Creates a new Flow Address and EVM account with a Cadence Owned Account (COA) stored in the account's storage.
+///
+transaction(
+    key: String,  // key to be used for the account
+    signatureAlgorithm: UInt8, // signature algorithm to be used for the account
+    hashAlgorithm: UInt8, // hash algorithm to be used for the account
+    weight: UFix64, // weight to be used for the account
+) {
+    let auth: auth(Storage, Keys, Capabilities) &Account
+
+    prepare(signer: auth(Storage, Keys, Capabilities) &Account) {
+        pre {
+            signatureAlgorithm >= 1 && signatureAlgorithm <= 3:
+                "Cannot add Key: Must provide a signature algorithm raw value that corresponds to "
+                .concat("one of the available signature algorithms for Flow keys.")
+                .concat("You provided ").concat(signatureAlgorithm.toString())
+                .concat(" but the options are either 1 (ECDSA_P256), 2 (ECDSA_secp256k1), or 3 (BLS_BLS12_381).")
+            hashAlgorithm >= 1 && hashAlgorithm <= 6:
+                "Cannot add Key: Must provide a hash algorithm raw value that corresponds to "
+                .concat("one of of the available hash algorithms for Flow keys.")
+                .concat("You provided ").concat(hashAlgorithm.toString())
+                .concat(" but the options are 1 (SHA2_256), 2 (SHA2_384), 3 (SHA3_256), ")
+                .concat("4 (SHA3_384), 5 (KMAC128_BLS_BLS12_381), or 6 (KECCAK_256).")
```

Comment:
Although Cadence supports many signature and hash algorithms, not all of them can be used to create a Flow account. 

The check here should be updated to include only the supported algorithms by a Flow account:
 - signature: `ECDSA_P256` and `ECDSA_secp256k1`
 - hash: `SHA2_256` and `SHA3_256`
 Any combination of the above works.
 
I checked our cadence docs and [this page](https://cadence-lang.org/docs/language/accounts/keys) seems to miss this info.
The info is available [here](https://developers.flow.com/build/basics/accounts#signature-and-hash-algorithms) but that's not the Cadence doc (the algorithms codes on the table are the access API ones not the Cadence ones, so you shouldn't use them)

Reply:
Got it